flask/sessions.py

```diff
@@ -332,6 +332,8 @@ def open_session(self, app, request):
 
     def save_session(self, app, session, response):
         domain = self.get_cookie_domain(app)
+        if domain != None:
+            is_ip(domain, "SESSION_COOKIE_DOMAIN", self)
```

Comment:
Why are you passing `self` here? This doesn't even work. Please test your code (and flask's unittests) before committing/pushing...